app/src/main/java/cm/aptoide/pt/store/view/GridStoreMetaWidget.java

```diff
@@ -90,279 +94,389 @@ public GridStoreMetaWidget(View itemView) {
+      secondaryName.setText(secondaryNameString);
+    }
   }
 
-  private void setupSecondaryInfo(Context context, String name, String secondaryIconUrl) {
-    secondaryName.setText(name);
-    ImageLoader.with(context)
-        .loadWithShadowCircleTransform(secondaryIconUrl, secondaryIcon);
+  private void showMainName(String mainNameString) {
+    if (mainNameString != null) {
+      mainName.setText(mainNameString);
+    }
   }
 
-  private void setupMainInfo(String name, StoreTheme theme, Context context, long followersCount,
-      long followingCount, String mainIconUrl, @DrawableRes int defaultMainIcon,
-      @DrawableRes int mainNameDrawble) {
-    setupMainInfo(name, theme, context, 0, followersCount, followingCount, false, mainIconUrl,
-        defaultMainIcon, mainNameDrawble);
+  private Observable<HomeMeta> createHomeMeta(GridStoreMetaDisplayable displayable) {
+    return displayable.isFollowingStore(storeAccessor)
+        .flatMap(isFollowing -> displayable.isStoreOwner(accountManager)
+            .map(isOwner -> new HomeMeta(displayable.getMainIcon(), displayable.getSecondaryIcon(),
+                displayable.getMainName(), displayable.getSecondaryName(), isOwner,
+                displayable.hasStore(), isFollowing, displayable.getSocialLinks(),
+                displayable.getAppsCount(), displayable.getFollowersCount(),
+                displayable.getFollowingsCount(), displayable.getDescription(),
+                getColorOrDefault(displayable.getStoreTheme()), displayable.getStoreId(),
+                displayable.hasStore(), displayable.getBadge())));
   }
 
-  private void setupTheme(StoreTheme theme, Context context) {
-    @ColorInt int color = getColorOrDefault(theme, context);
-    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
-      Drawable d = context.getDrawable(R.drawable.dialog_bg_2);
-      d.setColorFilter(color, PorterDuff.Mode.SRC_IN);
-      containerLayout.setBackground(d);
+  private @ColorInt int getColorOrDefault(StoreTheme theme) {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+      return getContext().getResources()
+          .getColor(theme.getPrimaryColor(), getContext().getTheme());
     } else {
-      Drawable d = context.getResources()
-          .getDrawable(R.drawable.dialog_bg_2);
-      d.setColorFilter(color, PorterDuff.Mode.SRC_IN);
-      containerLayout.setBackgroundDrawable(d);
+      return getContext().getResources()
+          .getColor(theme.getPrimaryColor());
     }
-    followStoreButton.setTextColor(color);
-    editStoreButton.setTextColor(color);
   }
 
-  private int getColorOrDefault(StoreTheme theme, Context context) {
-    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-      return context.getResources()
-          .getColor(theme.getPrimaryColor(), context.getTheme());
+  private void showMainIcon(String mainIconUrl) {
+    ImageLoader.with(getContext())
+        .loadWithShadowCircleTransform(mainIconUrl, mainIcon);
```

Comment:
Question: In a user has no store but has username and NO avatar, wouldn't this crash? Is this scenario even possible?